client/my-sites/sharing/connections/services/google-my-business.js

```diff
@@ -25,40 +32,62 @@ export class GoogleMyBusiness extends SharingService {
 	};
 
 	// override `createOrUpdateConnection` to ignore connection update, this is only useful for publicize services
-	createOrUpdateConnection = () => {};
+	createOrUpdateConnection = ( keyringConnectionId, externalUserId = 0 ) => {
+		this.props.saveSiteSettings( this.props.siteId, {
+			gmb_api_token: keyringConnectionId,
+			gmb_location_id: externalUserId,
+		} );
+	};
 
 	// override `removeConnection` to remove the keyring connection instead of the publicize one
 	removeConnection = () => {
 		this.setState( { isDisconnecting: true } );
-		this.props.deleteStoredKeyringConnection( last( this.props.keyringConnections ) );
+		this.props
+			.saveSiteSettings( this.props.siteId, {
+				gmb_api_token: null,
+				gmb_location_id: null,
+			} )
+			.then( () => {
+				this.setState( { isDisconnecting: false } );
+			} );
 	};
 
-	componentWillReceiveProps( { availableExternalAccounts } ) {
-		if ( ! isEqual( this.props.availableExternalAccounts, availableExternalAccounts ) ) {
-			this.setState( {
-				isConnecting: false,
-				isDisconnecting: false,
-			} );
+	componentWillMount() {
+		this.requestSettings( this.props );
+	}
+
+	requestSettings( props ) {
+		const { requestingSiteSettings, siteId } = props;
+		if ( ! requestingSiteSettings && siteId ) {
+			props.requestSiteSettings( siteId );
+		}
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( nextProps.siteId && this.props.siteId !== nextProps.siteId ) {
+			this.requestSettings( nextProps );
 		}
 
-		if ( ! this.state.isAwaitingConnections ) {
+		if ( this.state.isAwaitingConnections ) {
+			this.setState( {
+				isAwaitingConnections: false,
+				isSelectingAccount: true,
+			} );
 			return;
 		}
 
-		this.setState( {
-			isAwaitingConnections: false,
-			isRefreshing: false,
-		} );
+		if ( ! isEqual( this.props.brokenConnections, nextProps.brokenConnections ) ) {
```

Comment:
Not sure if this is needed here, we would need to test the broken state and the refresh flow to make sure